packages/buidler-core/src/internal/sentry/anonymizer.ts

```diff
@@ -94,7 +94,61 @@ export class Anonymizer {
   public anonymizeErrorMessage(errorMessage: string): string {
     // the \\ before path.sep is necessary for this to work on windows
     const pathRegex = new RegExp(`\\S+\\${path.sep}\\S+`, "g");
-    return errorMessage.replace(pathRegex, "<user-file>");
+
+    // for files that don't have a path separator
+    const fileRegex = new RegExp("\\S+\\.(js|ts)\\S*", "g");
+
+    // hide hex strings of 20 chars or more
+    const hexRegex = /(0x)?[0-9A-Fa-f]{20,}/g;
+
+    return errorMessage
+      .replace(pathRegex, ANONYMIZED_FILE)
+      .replace(fileRegex, ANONYMIZED_FILE)
+      .replace(hexRegex, (match) => match.replace(/./g, "x"));
+  }
+
+  public raisedByBuidler(event: Event): boolean {
```

Comment:
This function is messy. One alternative to all of this is to add a property "userFile" to each frame, but I don't know if that could cause problems with Sentry's dashboard.